### PR TITLE
Fixed the bug of missing value of password field

### PR DESCRIPTION
**Pull request by fyery-chen**
Problem:
When upgrading a cluster, the value of norman-password-type field will
be reset to blank.

Solution:
Set old value it it doesn't exist in data

Issue:
https://github.com/rancher/rancher/issues/19009

### CODE REVIEW

File: pkg/api/store/cluster/cluster_store.go [hunk 1]
```diff
@@ -405,3 +406,30 @@ func setBackupConfigSecretKeyIfNotExists(oldData, newData map[string]interface{}
...
+	}
+
+	var updatedConfig []map[string]interface{}
+	for _, newConfig := range newSlice {
+		if newConfig["password"] != nil {
+			updatedConfig = append(updatedConfig, newConfig)
+			continue
+		}
+		for _, oldConfig := range oldSlice {
+			if newConfig["url"] == oldConfig["url"] && oldConfig["password"] != nil {
```
Comment (orangedeng) [inline pkg/api/store/cluster/cluster_store.go]: I think we need to compare `user` as well.

Comment (fyery-chen) [reply pkg/api/store/cluster/cluster_store.go]: Updated！

**Comment by orangedeng** [comment]
LGTM

**Comment by cjellick** [comment]
lgtm

**Pull request merged**